File: server/src/models/item.ts
import { Schema, Types, Connection } from 'mongoose';

export type Scan = {
  scannedAt: Date;
  description: string;
  coordinates?: string;
};

export type Item = {
  _id?: string;
  userId: string;
  name: string;
  createdAt: Date;
  description?: string;
  contact?: string;
  scanned?: number;
  found?: boolean;
  scans?: Scan[];
};

const scan = {
  scannedAt: { type: Date, required: true },
  description: { type: String, require: true },
  coordinates: String,
};

const itemSchema = new Schema<Item>({
  _id: Types.ObjectId,
  userId: { type: String, required: true },
  name: { type: String, required: true },
  createdAt: { type: Date, required: true },
  description: String,
  contact: String,
  scanned: Number,
  found: Boolean,
  scans: [scan],
});

export const itemModelFactory = (conn: Connection) =>
  conn.model<Item>('Item', itemSchema);

File: server/Matdo.Web/Models/Item.cs
using System.ComponentModel.DataAnnotations;
using MongoDB.Bson;

namespace Matdo.Web.Models;

public class Item
{
    public ObjectId Id { get; set; }

    [Required]
    public string UserId { get; set; }

    [Required]
    public string Name { get; set; }

    public DateTime CreatedAt { get; set; }

    public string? Description { get; set; }
}
